.github/workflows/deploy-storybook.yml

```diff
@@ -0,0 +1,53 @@
+# Simple workflow for deploying static content to GitHub Pages
+name: Deploy Storybook to GH Pages
+
+on:
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+  workflow_call: 
+
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+# Allow one concurrent deployment
+concurrency:
+  group: "pages"
+  cancel-in-progress: true
+
+jobs:
+  deploy:
+    name: Deploy Storybook to GH Pages
+    runs-on: ubuntu-latest
+
+    environment:
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '18.12.x'
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Build Storybook
+        run: npm run build-storybook
+
+      - name: Setup Pages
+        uses: actions/configure-pages@v2
+
+      - name: Upload artifact
+        uses: actions/upload-pages-artifact@v1
+        with:
+          path: './docs'
+
+      - name: Deploy to GitHub Pages
+        id: deployment
+        uses: actions/deploy-pages@v1
```

Comment:
Would you be able to explain how this works? Our docs (GH Pages) are served from `./docs` on main. How does this action deploy docs without having write access to main?

Reply:
For sure! 



GH provides two ways to _deploy_ using GH pages:
1. **From a branch**: Directly consumes all files right from the branch content - given the correct path.
2. **From GH Actions**: We're required to manually upload our files to GH.

<img  src="https://github.com/Constructor-io/constructorio-ui-autocomplete/assets/29166076/3ff368bc-de6b-45f1-8d52-2cd189b5cad2" height="200" />

We can perform this upload using the `actions/upload-pages-artifact` action. As described in this [GitHub article](https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site#publishing-with-a-custom-github-actions-workflow).

You can also see a working example over on [this repo](https://github.com/Constructor-io/connect-template-toolkit/blob/main/.github/workflows/deploy-docs-to-gh-pages.yml).

Finally, answering your question. We simply don't need to change any files on the `main` branch, GH stores all resulting built files as artifacts directly consumed by GH Pages.